src/transformers/pipelines/conversational.py

```diff
@@ -173,6 +190,29 @@ def __init__(self, min_length_for_response=32, *args, **kwargs):
 
         self.min_length_for_response = min_length_for_response
 
+    def _get_history(self, conversation):
+        """
+        Private function (subject to change) that simply tokenizes and concatenates past inputs. Also saves that
+        tokenization into the conversation state.
+
+        Args:
+            conversation (:class:`~transformers.Conversation`)
+
+        Returns:
+            :obj:`List[int]`: The list of tokens for the past input of that conversation.
+        """
+        history = conversation._history[:]
+        index = conversation._index
+        for i, (past_user_input, generated_response) in enumerate(
+            zip(conversation.past_user_inputs[index:], conversation.generated_responses[index:])
+        ):
+            for el in (past_user_input, generated_response):
+                new_history = self._parse_and_tokenize([el])[0]
+                history.extend(new_history)
+            conversation._index = i + index + 1
+        conversation._history = history
+        return history[:]
```

Comment:
```suggestion
        return history
```

Reply:
Same here, and even more important, don't pass back a reference to users to cached values. They can do whatever they want with the copy, if it's a reference then they might mess it up:
```python
history = conversation._get_history()
history.append("a")  # Now conversation._history also contains "a" if I'm not mistaken
```